mathesar/api/ui/viewsets/version.py

```diff
@@ -0,0 +1,78 @@
+import requests
+from requests import ConnectionError
+
+from rest_framework import viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+import mathesar
+# TODO this import path is pretty long; might want to remove redundant occurances of "_exceptions"
+from mathesar.api.exceptions.version_exceptions.base_exceptions import GithubReleasesAPIException
+from mathesar.api.exceptions.generic_exceptions.base_exceptions import NetworkException
+
+
+class VersionViewSet(viewsets.ViewSet):
+
+    @action(methods=['get'], detail=False)
+    def current(self, _):
+        current_version = mathesar.__version__
+        release_info, status_code = _get_release_info_from_github_by_tag_name(
+            current_version
+        )
+        return Response(release_info, status=status_code)
+
+    @action(methods=['get'], detail=False)
+    def latest(self, _):
+        latest_version = _get_latest_release_info_from_github()
+        return Response(latest_version)
+
+
+def _get_release_info_from_github_by_tag_name(tag_name):
+    """
+    Returns a tuple of (release info json, github api response status code). The resulting response
+    is meant to use Github API's response code.
+
+    Will not panic in case of failure querying Github's APIs, because we can still provide the
+    version/tag_name.
+    """
+    def release_info_with_only_the_tag_name():
+        """
+        Follows the same json schema as returned by GH, except that only the tag_name key is set.
+        """
+        return dict(tag_name=tag_name)
+    url = f'{_release_api_base_url}/tags/{tag_name}'
+    try:
+        response = requests.get(url, _github_request_headers)
```

Comment:
What is the purpose of getting the information from github if we can generate it even in case of a Github response error

Reply:
According to the design spec, we need the tag name, link to release notes and a release date. If we can't access GitHub's APIs, we only know the tag name. Maybe not even that, since we don't guarantee that our version string will match a tag name.